### PR TITLE
Add all buildings to building hours

### PR DESCRIPTION
**Pull request by zimmer18**
Add all building hours on campus and their corresponding pictures.

### CODE REVIEW

**Review by elijahverdoorn — APPROVED** [review]
Looks good.

File: data/building-hours.json
Comment (hawkrives) [inline data/building-hours.json]: The indentation on these doesn't match the rest of the file.

File: data/building-hours.json
Comment (hawkrives) [inline data/building-hours.json:85]: Thanks for catching that flip.

File: data/building-hours.json
Comment (hawkrives) [inline data/building-hours.json:103]: This is why I like trailing commas. Sadly, JSON doesn't support them.

File: data/building-hours.json
Comment (hawkrives) [inline data/building-hours.json:176]: Why did the indent change?

File: data/building-hours.json
Comment (hawkrives) [inline data/building-hours.json:189]: Indentation

File: data/building-hours.json
Comment (hawkrives) [inline data/building-hours.json:204]: Change these to `0:00`, maybe? I don't know that I've ever seen a 24-hour clock that hits 24:00; they usually go 23:59 to 0:00.

But I might be wrong. I haven't actually done the research.

File: data/building-hours.json
Comment (hawkrives) [inline data/building-hours.json:219]: Same 24-hour comment, s/24:00/0:00

File: data/building-hours.json
Comment (hawkrives) [inline data/building-hours.json:234]: s/24:00/0:00

File: data/building-hours.json
Comment (hawkrives) [inline data/building-hours.json]: Christiansen Hall of Music? Or is that too long?

File: data/building-hours.json
Comment (hawkrives) [inline data/building-hours.json:249]: s/24:00/0:00

File: data/building-hours.json
Comment (hawkrives) [inline data/building-hours.json:264]: s/24:00/0:00

File: data/building-hours.json
Comment (hawkrives) [inline data/building-hours.json]: Yeah it confuses with this one…

File: data/building-hours.json
Comment (hawkrives) [reply data/building-hours.json]: Or at least `Christiansen`, or maybe `Christiansen (CHM)`

File: data/building-hours.json
Comment (hawkrives) [inline data/building-hours.json:309]: s/24:00/0:00

File: data/building-hours.json
Comment (hawkrives) [inline data/building-hours.json:324]: s/24:00/0:00

File: data/building-hours.json
Comment (hawkrives) [inline data/building-hours.json:339]: s/24:00/0:00

File: data/building-hours.json
Comment (hawkrives) [inline data/building-hours.json:354]: s/24:00/0:00

File: views/building-hours/index.js
Comment (hawkrives) [inline views/building-hours/index.js:25]: Two notes: One, I _had_ explicitly renamed all of these to `.jpeg`… 😉 but I don't really care, and I think I like `.jpg` better.

Two, the `.small` copy was a shrunken version of the big version, because no devices need to load a 1000-px-wide image for portrait mode, and it was making my Android simulators and phone laggy when I opened that view. I can change that after we merge; I'll run them through ImageOptim anyway, like I always do when a new image is included.

**Review by hawkrives — CHANGES_REQUESTED** [review]
Looks great! I'm glad you didn't listen to me before 😄 … since you've actually gone and done this, and my idea is still just an idea.

I've left some notes. I can be debated on some, but I like my indentations to match.

File: data/building-hours.json
Comment (elijahverdoorn) [reply data/building-hours.json]: I agree, this should be normalized.

File: data/building-hours.json
Comment (elijahverdoorn) [reply data/building-hours.json:103]: If only...

File: data/building-hours.json
Comment (elijahverdoorn) [reply data/building-hours.json:204]: It works this way too, I think it's not a big deal, I'd be ok either way. I think that the `24:00` is a little more descriptive for the programmer, semantically since it's larger than the `8:00`. I'll leave it up to you, @zimmer18

**Comment by elijahverdoorn** [comment]
I encouraged this work, @hawkrives, not with the intention of stepping on your ideas, but for the sake of getting it started and out the door. As it sits right now this system works, but if you want to go back later and implement your changes I'm not opposed and would like to be involved in that process (if we care enough to do it).

File: data/building-hours.json
Comment (elijahverdoorn) [reply data/building-hours.json]: I like `Christiansen (CHM)`

File: data/building-hours.json
Comment (drewvolz) [inline data/building-hours.json]: Not that the logic matters, but while we are being picky, would you mind flipping the order so that `Wed` and `Tue` switch spots, please?

File: data/building-hours.json
Comment (elijahverdoorn) [reply data/building-hours.json]: This is a usual point of confusion, since HOM and CHM are different.

File: data/building-hours.json
Comment (hawkrives) [reply data/building-hours.json:204]: I've been planning to remove the need for `nextDay` by the simple heuristic of "does it close before it opens", and that's a tiny bit trickier with a larger number. But I don't care either way.

File: views/building-hours/index.js
Comment (elijahverdoorn) [reply views/building-hours/index.js:25]: Yeah, I had to look at the Blame to be sure that these changes were you. I agree that a shrunken version is fine, and better for speed/download size. Thanks for offering to run them!

**Comment by hawkrives** [comment]
@elijahverdoorn That's why I'm glad you encouraged it – because now this is done, and mine has made no progress since mid-october. I'll be sure to drop into the all-about-olaf channel when I get back around to this screen – my current project is making a new FancyMenu view based on what Matt started.

File: data/building-hours.json
Comment (hawkrives) [reply data/building-hours.json]: Maybe `Christiansen (CHM)` and `Hall of Music (HOM)`? Put the acronyms on both?

File: views/building-hours/index.js
Comment (zimmer18) [reply views/building-hours/index.js:25]: I made all of the images 1440x540px which was the same size as the image.small.jpeg. I'm not sure if that's what ImageOptim would do, but I thought you should know that @hawkrives .

Comment (hawkrives) [reply views/building-hours/index.js:25]: If you still have the originals, could you add them as well, please? 

ImageOptim is a nifty tool that reduces image filesize losslessly by primarily removing image metadata. It's a Mac app, but pretty handy.

Comment (hawkrives) [reply views/building-hours/index.js:25]: But I won't block this PR on the original side being included. No phone goes above ~900px right now, anyway, so I should shrink them some more already.

Comment (zimmer18) [reply views/building-hours/index.js:25]: Sounds good! I'll add them.

**Review by drewvolz — APPROVED** [review]

**Comment by zimmer18** [comment]
@hawkrives I still need to add the old pictures.

**Review by hawkrives — APPROVED** [review]

**Comment by hawkrives** [comment]
👍

**Pull request merged**
